packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -315,56 +315,51 @@ fn mem_batch_to_js<'m>(
     mv8: &'m MiniV8,
     batch_id: &str,
     mem: mv8::Object<'m>,
-    metaversion: &Metaversion,
+    persisted: Metaversion,
 ) -> Result<mv8::Value<'m>> {
     let batch = mv8.create_object();
     let batch_id = mv8.create_string(batch_id);
     batch.set("id", mv8::Value::String(batch_id))?;
     batch.set("mem", mem)?;
-    batch.set("mem_version", metaversion.memory())?;
-    batch.set("batch_version", metaversion.batch())?;
+    batch.set("mem_version", persisted.memory())?;
+    batch.set("batch_version", persisted.batch())?;
     Ok(mv8::Value::Object(batch))
 }
 
 fn batch_to_js<'m>(
     mv8: &'m MiniV8,
     mem: &Memory,
-    metaversion: &Metaversion,
+    persisted: Metaversion,
 ) -> Result<mv8::Value<'m>> {
     // TODO: Is `mem.data.len()` different from `mem.size`? (like Vec capacity vs len?)
     let arraybuffer = mv8.create_arraybuffer(mem.data.as_ptr(), mem.size);
-    let batch_id = mem.get_id();
-    mem_batch_to_js(mv8, batch_id, arraybuffer, metaversion)
-}
-
-fn _mut_batch_to_js<'m>(
-    mv8: &'m MiniV8,
-    mem: &mut Memory,
-    metaversion: &Metaversion,
-) -> Result<mv8::Value<'m>> {
-    // TODO: Is `mem.data.len()` different from `mem.size`?
-    let arraybuffer = mv8.create_arraybuffer(mem.as_mut_ptr(), mem.size);
-    let batch_id = mem.get_id();
-    mem_batch_to_js(mv8, batch_id, arraybuffer, metaversion)
+    let batch_id = mem.id();
+    mem_batch_to_js(mv8, batch_id, arraybuffer, persisted)
 }
 
 fn state_to_js<'m, 'a, 'b>(
     mv8: &'m MiniV8,
     agent_batches: impl Iterator<Item = &'a AgentBatch>,
-    msg_batches: impl Iterator<Item = &'b MessageBatch>,
+    message_batches: impl Iterator<Item = &'b MessageBatch>,
 ) -> Result<(mv8::Value<'m>, mv8::Value<'m>)> {
     let js_agent_batches = mv8.create_array();
     let js_message_batches = mv8.create_array();
 
-    for (i_batch, (agent_batch, message_batch)) in agent_batches
-        .into_iter()
-        .zip(msg_batches.into_iter())
-        .enumerate()
+    for (i_batch, (agent_batch, message_batch)) in
+        agent_batches.into_iter().zip(message_batches).enumerate()
```

Comment:
zip_eq?

Reply:
Yes! But sadly, there is no `zip_eq` function on iterators. Added an assertion to ensure, both iterators had the same length.

Reply:
Resolving this and commenting on the new commit